XenModel/XenAPI-Extensions/Session.cs

```diff
@@ -78,13 +79,17 @@ public Session(Session session, Proxy proxy, IXenConnection connection)
         /// </summary>
         public Session(Session session, IXenConnection connection, int timeout)
         {
-            if (session.JsonRpcClient != null)
+            APIVersion = session.APIVersion;

```

Comment:
I think this constructor can call the existing Session(Session session, int timeout). This way we would avoid duplicated code.